server/src/services/Google.ts

```diff
@@ -189,98 +185,166 @@ function getStandardPatchBody({
     guestsCanSeeOtherGuests: false,
     guestsCanInviteOthers: false,
   };
-  return body;
 }
 
-export async function createCalendarEvent(eventData: EventData) {
+interface CalendarId {
+  calendarId: string;
+}
+
+export async function createCalendarEvent(
+  { calendarId }: CalendarId,
+  eventData: EventData,
+) {
   const calendarApi = await createCalendarApi();
 
-  const { calendarId } = eventData;
   const { data } = await calendarApi.events.insert({
     calendarId,
     sendUpdates: 'all',
-    requestBody: getStandardRequestBody(eventData),
+    requestBody: createEventRequestBody(eventData),
   });
 
   return { calendarEventId: data.id };
 }
 
-// TODO: create a patchCalendarEvent for updating specific fields (most useful
-// for RSVPs which only modify the attendees list)
-export async function updateCalendarEvent(eventUpdateData: EventUpdateData) {
+interface EventIds {
+  calendarId: string;
+  calendarEventId: string;
+}
+
+async function getAndUpdateEvent(
+  { calendarId, calendarEventId: eventId }: EventIds,
+  update: EventData | null,
+  updateAttendees?: (
+    attendees?: calendar_v3.Schema$EventAttendee[],
+  ) => calendar_v3.Schema$EventAttendee[] | undefined,
+) {
   const calendarApi = await createCalendarApi();
 
-  const { calendarId, calendarEventId } = eventUpdateData;
+  const { data } = await calendarApi.events.get({
+    calendarId,
+    eventId,
+  });
+
+  const { attendees } = data;
+
+  const updatedAttendeesData = updateAttendees
+    ? updateAttendees(attendees)
+    : attendees;
+
+  const requestBodyUpdates: EventData = {
+    ...update,
+    ...{ attendees: updatedAttendeesData },
+  };
+
   await calendarApi.events.update({
     calendarId,
-    eventId: calendarEventId,
+    eventId,
     sendUpdates: 'all',
-    requestBody: getStandardRequestBody(eventUpdateData),
+    requestBody: { ...data, ...createEventRequestBody(requestBodyUpdates) },
   });
 }
 
-export async function patchCalendarEvent(
-  eventUpdateData: Partial<EventUpdateData>,
+// To be used to update event, but not the attendees.
+export async function updateCalendarEvent(
+  { calendarId, calendarEventId }: EventIds,
+  eventUpdateData: EventData,
 ) {
-  const calendarApi = await createCalendarApi();
+  await getAndUpdateEvent({ calendarId, calendarEventId }, eventUpdateData);
+}
 
-  const { calendarId, calendarEventId } = eventUpdateData;
+// TODO: use alias for id and ids
+export async function cancelCalendarEvent({
+  calendarId,
+  calendarEventId,
+}: EventIds) {
+  await getAndUpdateEvent(
+    { calendarId, calendarEventId },
+    { status: 'cancelled' },
+  );
+}
 
-  await calendarApi.events.patch({
-    calendarId,
-    eventId: calendarEventId,
-    sendUpdates: 'all',
-    requestBody: getStandardPatchBody(eventUpdateData),
-  });
+function filterFromAttendees(email: string) {
+  return (attendees: calendar_v3.Schema$EventAttendee[]) =>
+    attendees.filter((attendee) => attendee.email === email);
```

Comment:
Are we leaving here other attendees, or just the one having `email`?
```suggestion
    attendees.filter((attendee) => attendee.email !== email);
```

Reply:
Oops. It's what you suggested.

Looks like I'm writing those tests now!